compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -6276,7 +6276,7 @@ object Types extends TypeUtils {
      */
     def expandParam(tp: NamedType, pre: Type): Type =
       tp.argForParam(pre) match {
-        case arg @ TypeRef(pre, _) if pre.isArgPrefixOf(arg.symbol) =>
+        case arg @ TypeRef(`pre`, _) if pre.isArgPrefixOf(arg.symbol) =>
```

Comment:
Hah, we really need a shadowing warning for this.

Reply:
Yeah, probably.  Obviously it's worst when the types match up exactly, but I've definitely shadowed things intentionally, ala `port.parse.match { case Some(port) => ... }`, so hard to find the right ergonomics...